lib/util/cache/repository/types.ts

```diff
@@ -27,14 +27,38 @@ export interface BranchUpgradeCache {
 }
 
 export interface BranchCache {
+  /**
+   *Whether this branch has automerge enabled
+   */
   automerge: boolean;
+  /**
+   * Hash of the manager fingerprints and the update branch config
+   */
+  branchFingerprint?: string;
+  /**
+   * Branch name
+   */
   branchName: string;
+  /**
+   * Whether a person not listed in gitIgnoredAuthors updated the branch.
+   */
   isModified: boolean;
+  /**
+   * Parent commit of branch sha
```

Comment:
is this the commit Sha when the branch was created from base branch?

Reply:
Yes and no.
When the branch was first created "yes". But if the branch was modified after that then it is the 2nd last commit.
Ie. 
Branch Created and has sha as S1, parentSha as P1
Now user or bot makes another commit, sha becomes S2 and parentSha becomes S1.

Reply:
What's the reason for *updating* this over time?

Reply:
Not sure what you mean to ask.

Reply:
If there's a new commit, what's the value/reason for updating `parentSha` to value `S1`? Do we use that value anywhere?

Reply:
You metioned that you use it in the app-backend to compute mergeConfidence

Reply:
Also it gets calculated likt this each run:
```
 parentSha = await git.revparse([`${branchSha!}^`]);
```
So if we want to use cached value on re-runs we need to update its value in cache. Else while setting cache we will end up cloning again.

Reply:
ok, so this is wrong in all cases when there are commits ontop of the renovate branch?
I think we need to fix this too. it should probably always point to the renovate commit (eg first branch commit)

Reply:
Not sure. @rarkins thoughts??

Reply:
@RahulGautamSingh please create an issue to track this, so we can discuss it separately